duoapi.go

```diff
@@ -288,6 +343,71 @@ func (duoapi *DuoApi) SignedCall(method string,
 	return duoapi.makeRetryableHttpCall(method, url, headers, requestBody, options...)
 }
 
+type JSONParams map[string]interface{}
+
+// Make a signed Duo Rest API call that takes JSON as an argument.
+// See Duo's online documentation for the available REST API's.
+// method is one of GET, POST, PATCH, PUT, DELETE
+// uri is the URI of the Duo Rest call
+// json is the JSON parameters to include in the call.
+// options Optional parameters.  Use UseTimeout to toggle whether the
+// Duo Rest API call should timeout or not.
+//
+//	Example:
+//	params := duoapi.JSONParams{
+//		"user_id":         userid,
+//		"activation_code": activationCode,
+//	}
+//	JSONSignedCall("POST", "/auth/v2/enroll_status", params, duoapi.UseTimeout)
+func (duoapi *DuoApi) JSONSignedCall(method string,
+	uri string,
+	params JSONParams,
+	options ...DuoApiOption) (*http.Response, []byte, error) {
+
+	body_methods := make(map[string]struct{})
+	body_methods["POST"] = struct{}{}
+	body_methods["PUT"] = struct{}{}
+	body_methods["PATCH"] = struct{}{}
+	_, params_go_in_body := body_methods[method]
+
+	now := time.Now().UTC().Format(time.RFC1123Z)
+	var body string
+	api_url := url.URL{
+		Scheme: "https",
+		Host:   duoapi.host,
+		Path:   uri,
+	}
+
+	url_values := url.Values{}
+	if params_go_in_body {
+		body_bytes, _ := json.Marshal(params)
+		body = string(body_bytes[:])
+	} else {
+		body = ""
+		var err error
+		url_values, err = jsonToValues(params)
+		if err == nil {
```

Comment:
Can you change this check to just check for an error and if there isn't one to proceed with the Encode call
```
if err != nil {
    return nil, nil, err
}
api_url.RawQuery = url_values.Encode()
```
I think it reads easier as a default case.